### ISSUE TITLE
Bug: Executing AndroidTests fails because connectedDebugAndroidTest task fails to build

### ISSUE THREAD

**Issue by Marty-Byrde** [Marty-Byrde/Reengineering-Project#2]
Executing any test inside of the androidTest directory leads to a build-exception. 

```
> Task :compileDebugJavaWithJavac
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :connectedDebugAndroidTest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':connectedDebugAndroidTest'.
```

The current analysis shows that this Exception may be caused by an internal module that throws an `IllegalAccessError`.

<img width="1630" alt="image" src="https://github.com/user-attachments/assets/aef29831-e988-4be1-a806-d1e6af23328d">

**Comment by Marty-Byrde** [comment]
When all directories inside of the `src/androidTest` directory are removed, which means that all androidTests are removed. Then, the build process executes successfully (` ./gradlew connectedAndroidTest`). 

Adding an empty kotlin class at `src/androidTest/kotlin/Example` without any annotations already leads to a build failure.

**Comment by Marty-Byrde** [comment]
This issue was been resolved by #3  by increasing and thereby migrating to a higher `compileSDK` version.
